src/osg/Texture2D.cpp

```diff
@@ -297,30 +471,18 @@ void Texture2D::apply(State& state) const
     }
     else if ( (_textureWidth!=0) && (_textureHeight!=0) && (_internalFormat!=0) )
     {
+        textureObject = generateAndAssignTextureObject(contextID,GL_TEXTURE_2D,_numMipmapLevels,_internalFormat,_textureWidth,_textureHeight,1,_borderWidth);
+
+        textureObject->bind();
+
+        applyTexParameters(GL_TEXTURE_2D,state);
 
         // no image present, but dimensions at set so lets create the texture
-         GLExtensions * extensions = state.get<GLExtensions>();
```

Comment:
More code that can go back in now that selectSizedInternalFormat is going back into Texture.cpp